src/common_format.rs

```diff
@@ -31,34 +34,133 @@ pub struct CalendarDate {
+    #[serde(deserialize_with = "de_from_u8", serialize_with = "ser_from_bool")]
+    friday: bool,
+    #[serde(deserialize_with = "de_from_u8", serialize_with = "ser_from_bool")]
+    saturday: bool,
+    #[serde(deserialize_with = "de_from_u8", serialize_with = "ser_from_bool")]
+    sunday: bool,
+    #[serde(deserialize_with = "de_from_date_string", serialize_with = "ser_from_naive_date")]
+    start_date: Date,
+    #[serde(deserialize_with = "de_from_date_string", serialize_with = "ser_from_naive_date")]
+    end_date: Date,
+}
+
+impl Calendar {
+    fn get_valid_days(&self) -> Vec<chrono::Weekday> {
+        let mut valid_days: Vec<chrono::Weekday> = vec![];
+        if self.monday {
+            valid_days.push(chrono::Weekday::Mon);
+        }
+        if self.tuesday {
+            valid_days.push(chrono::Weekday::Tue);
+        }
+        if self.wednesday {
+            valid_days.push(chrono::Weekday::Wed);
+        }
+        if self.thursday {
+            valid_days.push(chrono::Weekday::Thu);
+        }
+        if self.friday {
+            valid_days.push(chrono::Weekday::Fri);
+        }
+        if self.saturday {
+            valid_days.push(chrono::Weekday::Sat);
+        }
+        if self.sunday {
+            valid_days.push(chrono::Weekday::Sun);
+        }
+
+        valid_days
+    }
+
+    fn get_valid_dates(&self) -> BTreeSet<Date> {
+        let valid_days = self.get_valid_days();
+        let duration = self.end_date - self.start_date;
+        (0..duration.num_days() + 1)
+            .map(|i| self.start_date + chrono::Duration::days(i))
+            .filter(|d| valid_days.contains(&d.weekday()))
+            .collect()
+    }
+}
+
+fn manage_calendar_dates(
+    calendars: &mut CollectionWithId<objects::Calendar>,
+    path: &path::Path,
+) -> Result<()> {
+    let file = "calendar_dates.txt";
+    let path = path.join(file);
+    if !path.exists() {
+        info!("Skipping {}", file);
+    } else {
+        info!("Reading {}", file);
+
+        let mut rdr = csv::Reader::from_path(&path).with_context(ctx_from_path!(path))?;
+        for calendar_date in rdr.deserialize() {
+            let calendar_date: CalendarDate = calendar_date.with_context(ctx_from_path!(path))?;
+
+            let is_inserted = calendars
+                .get_mut(&calendar_date.service_id)
+                .map(|mut calendar| match calendar_date.exception_type {
+                    ExceptionType::Add => {
+                        calendar.dates.insert(calendar_date.date);
+                    }
+                    ExceptionType::Remove => {
+                        calendar.dates.remove(&calendar_date.date);
+                    }
+                });
+            is_inserted.unwrap_or_else(|| {
+                if calendar_date.exception_type == ExceptionType::Add {
+                    let mut dates = BTreeSet::new();
+                    dates.insert(calendar_date.date);
+                    calendars.push(objects::Calendar {
```

Comment:
here I have a warning
```
warning: unused `std::result::Result` which must be used
   --> src/common_format.rs:128:21
    |
128 | /                     calendars.push(objects::Calendar {
129 | |                         id: calendar_date.service_id,
130 | |                         dates,
131 | |                     });
    | |_______________________^
    |
    = note: #[warn(unused_must_use)] on by default
```
I don't know how to remove it

Reply:
adding a `?` at the end.

Reply:
```
--> src/common_format.rs:128:21
    |
128 | /                     calendars.push(objects::Calendar {
129 | |                         id: calendar_date.service_id,
130 | |                         dates,
131 | |                     })?;
    | |_______________________^ cannot use the `?` operator in a function that returns `(
```

Reply:
works with unwrap as suggested by @prhod